lib/rack/utf8_sanitizer.rb

```diff
@@ -155,8 +163,8 @@ def decode_string(input)
     # However, the regexp approach used by URI.unescape is not sophisticated
     # enough for our task.
     def unescape_unreserved(input)
-      input.gsub(/%([a-f\d]{2})/i) do |encoded|
-        decoded = [$1.hex].pack('C')
+      input.gsub(PERCENT_CHAR_REGEXP) do |encoded|
+        decoded = [$1.hex].pack(PACK_C)
```

Comment:
I just noticed the array here too. It might be worth benchmarking `$1.hex.char` vs this?